client/van_serviceinterface_update.go

```diff
@@ -0,0 +1,355 @@
+		}
+	}
+	if !modified {
+		targets = append(targets, *target)
+	}
+	service.Targets = targets
+}
+
+func getServiceInterfaceTarget(targetType string, targetName string, deducePort bool, cli *VanClient) (*types.ServiceInterfaceTarget, error) {
+	if targetType == "deployment" {
+		deployment, err := cli.KubeClient.AppsV1().Deployments(cli.Namespace).Get(targetName, metav1.GetOptions{})
+		if err == nil {
+			target := types.ServiceInterfaceTarget {
+				Name: deployment.ObjectMeta.Name,
+				Selector: stringifySelector(deployment.Spec.Selector.MatchLabels),
+			}
+			if deducePort {
+				//TODO: handle case where there is more than one container (need --container option?)
+				if deployment.Spec.Template.Spec.Containers[0].Ports != nil {
+					target.TargetPort = int(deployment.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort)
+				}
+			}
+			return &target, nil
+		} else {
+			return nil, fmt.Errorf("Could not read deployment %s: %s", targetName, err)
+		}
+	} else if targetType == "statfulset" {
+		statefulset, err := cli.KubeClient.AppsV1().StatefulSets(cli.Namespace).Get(targetName, metav1.GetOptions{})
+		if err == nil {
+			target := types.ServiceInterfaceTarget {
+				Name: statefulset.ObjectMeta.Name,
+				Selector: stringifySelector(statefulset.Spec.Selector.MatchLabels),
+			}
+			if deducePort {
+				//TODO: handle case where there is more than one container (need --container option?)
+				if statefulset.Spec.Template.Spec.Containers[0].Ports != nil {
+					target.TargetPort = int(statefulset.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort)
+				}
+			}
+			return &target, nil
+		} else {
+			return nil, fmt.Errorf("Could not read statefulset %s: %s", targetName, err)
+		}
+	} else if targetType == "pods" {
+		return nil, fmt.Errorf("VAN service interfaces for pods not yet implemented")
+	} else {
+		return nil, fmt.Errorf("VAN service interface unsupported target type")
+	}
+}
+
+func updateServiceInterface(service *types.ServiceInterface, overwriteIfExists bool, owner *metav1.OwnerReference, cli *VanClient) error {
+	encoded, err := jsonencoding.Marshal(service)
+	if err != nil {
+		return fmt.Errorf("Failed to encode service interface as json: %s", err)
+	}
+	current, err := cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Get("skupper-services", metav1.GetOptions{})
+	if err == nil {
+		if overwriteIfExists || current.Data == nil || current.Data[service.Address] == "" {
+			if current.Data == nil {
+				current.Data = map[string]string{
+					service.Address: string(encoded),
+				}
+			} else {
+				current.Data[service.Address] = string(encoded)
+			}
+			_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Update(current)
+			if err != nil {
+				return fmt.Errorf("Failed to update skupper-services config map: %s", err)
+			} else {
+				return nil
+			}
+		} else {
+			return fmt.Errorf("Service %s already defined", service.Address)
+		}
+	} else if errors.IsNotFound(err) {
```

Comment:
Would it work to assume skupper-services cm is create during van_router_create and treat this as exceptional?

Reply:
Possibly, but I'd quite like to be able to handle e.g. that getting deleted and recreated.